Source/WebCore/crypto/SubtleCrypto.cpp

```diff
@@ -1107,12 +1107,13 @@ void SubtleCrypto::wrapKey(JSC::JSGlobalObject& state, KeyFormat format, CryptoK
     auto exportAlgorithm = CryptoAlgorithmRegistry::singleton().create(key.algorithmIdentifier());
     auto wrapAlgorithm = CryptoAlgorithmRegistry::singleton().create(wrappingKey.algorithmIdentifier());
 
-    auto context = scriptExecutionContext();
+    ScriptExecutionContext* context = scriptExecutionContext();
 
     auto index = promise.ptr();
     m_pendingPromises.add(index, WTFMove(promise));
     WeakPtr weakThis { *this };
-    auto callback = [index, weakThis, wrapAlgorithm, wrappingKey = Ref { wrappingKey }, wrapParams = WTFMove(wrapParams), isEncryption, context, workQueue = m_workQueue](SubtleCrypto::KeyFormat format, KeyData&& key) mutable {
+    bool useCryptoKit = context && context->settingsValues().cryptoKitEnabled;
+    auto callback = [index, weakThis, wrapAlgorithm, wrappingKey = Ref { wrappingKey }, wrapParams = WTFMove(wrapParams), isEncryption, context, workQueue = m_workQueue, useCryptoKit](SubtleCrypto::KeyFormat format, KeyData&& key) mutable {
```

Comment:
Not about your code, but we probably should't be capturing the raw pointer context.  Let's get it from weakThis when callback is called.  Also, early returns make less indentation:

if (!weakThis)
   return;
auto* context = weakThis->scriptExecutionContext();
if (!context)
    return;
RefPtr promise = weakThis->m_pendingPromises.get(index);
if (!promise)
    return;
...

Reply:
refactored! 
and yes early returns definitely make this blob easier to read!